Assets/Materials/PlayerLight.mat

```diff
@@ -0,0 +1,77 @@
+%YAML 1.1
+%TAG !u! tag:unity3d.com,2011:
+--- !u!21 &2100000
+Material:
```

Comment:
You used "Legacy Shaders/Transparent/Bumped Diffuse" for this material but the shader "Sprites/Diffuse" do the same and it is not legacy, please use this one.

Reply:
Ok so I did few experiments to have shadows and I find out that shadows only works with a material with the "Standard" shader.